MovingBlocks/TerasologyLauncher
# i18n: use properties and bindings for i18n of resources bundles

### Contains

- slim down `Languages` and merge it with `BundleUtils`
- rename `BundleUtils` >>> `I18N`, now holding utility for translating resource bundles in general
- add small FX-utility to `I18N` to create a `Tooltip` with a string binding
- migrate both application and settings controller to use string bindings for labels
- use `Locale`s as value for the language selection and use custom cell renderer for icon and translated text

Based on https://sodocumentation.net/javafx/topic/5434/internationalization-in-javafx

Contributes to #505.

### How to test

This binds the translation of labels and tooltips to the current settings. 

- note the current locale (e.g., check the tooltip for the settings button :gears:)
- go to settings
- change the language (language items should be rendered with icon and text)
  - [x] languages are correctly rendered
  - [x] the labels in the settings view update accordingly
- cancel the changes (this brings you back to the main view)
  - [x] the current locale hasn't change (again, tooltip of settings button)
- go to settings again
- change the language
  - [x] labels still update correctly
- click save to confirm the settings
  - [x] confirm the tooltips have updated
- close and restart the launcher
  - [x] the locale was persisted to the settings (`~/.terasologylauncher/settings.json`)  and is picked up on next start

For good measure, switch up the steps, close and restart in between, etc. 

Also, please take note of any labels or texts that do not update when changing the locale.

### Outstanding before merging

- [x] bind translation for `JavaHeapSize` or remove it completely? (see https://github.com/MovingBlocks/TerasologyLauncher/pull/679#issuecomment-1301278345)
- [x] check bindings and how they re set up, not sure whether I've created a (possible) knot somewhere...
